cmd/serv.go

```diff
@@ -297,19 +302,26 @@ func runServ(c *cli.Context) error {
 		gitcmd = exec.CommandContext(ctx, verb, repoPath)
 	}
 
-	// Check if setting.RepoRootPath exists. It could be the case that it doesn't exist, this can happen when
-	// `[repository]` `ROOT` is a relative path and $GITEA_WORK_DIR isn't passed to the SSH connection.
-	if _, err := os.Stat(setting.RepoRootPath); err != nil {
-		if os.IsNotExist(err) {
-			return fail("Incorrect configuration.",
-				"Directory `[repository]` `ROOT` was not found, please check if $GITEA_WORK_DIR is passed to the SSH connection or make `[repository]` `ROOT` an absolute value.")
-		}
-	}
-
 	gitcmd.Dir = setting.RepoRootPath
 	gitcmd.Stdout = os.Stdout
 	gitcmd.Stdin = os.Stdin
 	gitcmd.Stderr = os.Stderr
+	gitcmd.Env = append(gitcmd.Env, os.Environ()...) // FIXME: the legacy code passes the whole env. in the future, should only pass the env which are really needed
+	gitcmd.Env = append(gitcmd.Env,
+		repo_module.EnvRepoIsWiki+"="+strconv.FormatBool(results.IsWiki),
+		repo_module.EnvRepoName+"="+results.RepoName,
+		repo_module.EnvRepoUsername+"="+results.OwnerName,
+		repo_module.EnvPusherName+"="+results.UserName,
+		repo_module.EnvPusherEmail+"="+results.UserEmail,
+		repo_module.EnvPusherID+"="+strconv.FormatInt(results.UserID, 10),
+		repo_module.EnvRepoID+"="+strconv.FormatInt(results.RepoID, 10),
+		repo_module.EnvPRID+"="+fmt.Sprintf("%d", 0),
+		repo_module.EnvDeployKeyID+"="+fmt.Sprintf("%d", results.DeployKeyID),
+		repo_module.EnvKeyID+"="+fmt.Sprintf("%d", results.KeyID),
+		repo_module.EnvAppURL+"="+setting.AppURL,
+	)
+	gitcmd.Env = append(gitcmd.Env, git.CommonEnvs()...)
```

Comment:
> CommonEnvs includes:
> 
>     1. `LC_ALL`: which will override the users specific language
> 
>     2. `GIT_TERMINAL_PROMPT`: which I think is going to prevent the user from having the option to pass in commands
> 
> 
> I think `GIT_NO_REPLACE_OBJECTS` is probably reasonable and `GIT_CONFIG_NOSYSTEM` similarly.

For 1 `LC_ALL`: I prefer to only provide English response for consistency. If there is any error message from Gitea side, it's English-only here too. Non-English responses make maintainer difficult to see user's problem too.

For 2 `GIT_TERMINAL_PROMPT`: do you have an example for this case? I don't think the terminal prompt is needed during ssh/push. Even if it's needed, we can find the root case later and fix it in the future.

Reply:
Gitea's lack of managing `LC_ALL` and requested locales is a bug. It's not right to make the situation worse.

I'm not certain of a good way in which `GIT_TERMINAL_PROMPT` could be used but I don't see a good reason for switching it off either.

Reply:
Do you mean the `LC_ALL` and `GIT_TERMINAL_PROMPT` here must be removed?

Reply:
Yeah I don't think we should be importing CommonEnvs here at all. 

By all means set HOME & GIT_NO_REPLACE_OBJECTS=1 but the LC_ALL and GIT_TERMINAL_PROMPT settings to my mind seem like the wrong thing to do. Certainly the LC_ALL.  

I might be wrong here and whilst I agree an argument could be made to set LC_ALL temporarily here until we get the locales sorted in serv I just know that if we stick CommonEnvs in then when we do get locales sorted what's going to happen is someone is going to remove that from CommonEnvs and then things like merging error detection will break.

Reply:
Done. Now `gitea serv` uses `git.CommonCmdServEnvs`, and internal `git` command uses `git.CommonGitCmdEnvs`.

btw: I think the reason to set `LC_ALL` for all sub commands is: some Git UI clients may parse responded message for their UI and interaction. I guess these Git clients may only support English response messages for parsing, so it's good to set `LC_ALL` for the sub commands of `gitea serv`. Usually the messages read by machine do not need to be localized.